### PR TITLE
Feature/lego nxt

### PR DESCRIPTION
**Pull request by martinbu**
Remove old lego nxt implementation and added new one with sensor functionality

CAT-1182
CAT-1183
CAT-1383
CAT-1382
CAT-1371
CAT-1370

Tests: 
https://jenkins.catrob.at/view/All-Categories/view/Catroid-multi-job/job/Catroid-Multi-Job-Custom-Branch-RELOADED/1855/
https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/371/

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/content/Project.java [hunk 1]
```diff
@@ -48,11 +50,15 @@
 
 	@XStreamAlias("header")
 	private XmlHeader xmlHeader = new XmlHeader();
+
 	@XStreamAlias("objectList")
 	private List<Sprite> spriteList = new ArrayList<Sprite>();
 	@XStreamAlias("data")
 	private DataContainer dataContainer = null;
 
+	@XStreamAlias("settings")
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/content/Project.java]: '.xml' change!

File: catroid/src/org/catrobat/catroid/io/XStreamToSupportCatrobatLanguageVersion095AndBefore.java [hunk 2]
```diff
@@ -412,10 +414,34 @@ private void modifyXMLToSupportUnknownFields(File file) {
...
+			}
+		}
+
+		NodeList motorMoveBricks = originalDocument.getElementsByTagName(oldMotorMoveBrickName);
+		for (int i = 0; i < motorMoveBricks.getLength(); i++) {
+			Node motorMoveBrick = motorMoveBricks.item(i);
+			originalDocument.renameNode(motorMoveBrick, motorMoveBrick.getNamespaceURI(), newMotorMoveBrickName);
+		}
+	}
+
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/io/XStreamToSupportCatrobatLanguageVersion095AndBefore.java]: Nice! Those renames got handled correctly, but we still need to increment our Catrobat Language Version.
( + rename of the `XStreamToSupportCatrobatLanguageVersion095..` classes)

File: catroidTest/src/org/catrobat/catroid/test/io/XStreamToSupportCatrobatLanguageVersion095AndBeforeTest.java [hunk 3]
```diff
@@ -213,4 +215,16 @@ public void testLoadingProjectsOfCatrobatLanguageVersion095() {
 		TestUtils.deleteTestProjects(PROJECT_NAME_GHOST_EFFECT_BRICKS);
 	}
 
+	public void testLoadingLegoNxtProjectsOfCatrobatLanguageVersion092() {
```
Comment (obusher) [inline catroidTest/src/org/catrobat/catroid/test/io/XStreamToSupportCatrobatLanguageVersion095AndBeforeTest.java]: ~ `testLoadingProjectsOfCatrobatLanguageVersion096()`

**Comment by obusher** [comment]
Why got those `/res/values-xx/strings.xml` touched ?
`

**Comment by martinbu** [comment]
1) XML change: Yes the xml got changed, but the language version was already increased to 0.95 . And this had be done only once for a release. Wolfgang Slany: "Erhöht wird immer nur für offizielle externe Versionen."

2) strings.xml: because string (variable name) got renamed. If you only rename it in the default xml you get checkstyle errors, that there are translations which are not present in the default xml. Also strings got deleted.

**Pull request merged**